### PR TITLE
Fi client delets all relationships

### PR DESCRIPTION
**Pull request by arkowalski**
added funcioanlity that does not use arn to delete all relationships for given nino and service name("afi")
added tests and adjusted the audit
readme.md updated

requires squashing commits

### CODE REVIEW

**Comment by arkowalski** [comment]
commits needs squashing, so if anything ask me

File: test/uk/gov/hmrc/agentfirelationship/controllers/RelationshipControllerSpec.scala [hunk 1]
```diff
@@ -40,17 +40,19 @@ class RelationshipControllerSpec extends UnitSpec with MockitoSugar with GuiceOn
...
   }
 
   private def authStub(returnValue: Future[~[Option[AffinityGroup], Enrolments]]) =
-    when(mockPlayAuthConnector.authorise(any(), any[Retrieval[~[Option[AffinityGroup], Enrolments]]]())(any(), any())).thenReturn(returnValue)
+    when(mockPlayAuthConnector.authorise(any(), any[Retrieval[~[Option[AffinityGroup],
+      Enrolments]]]())(any(), any())).thenReturn(returnValue)
 
   "RelationshipController" should {
-    "return Status: OK when successfully finding a relationship" in {
+    "return Status: OK when successfully finding relationship" in {
```
Comment (Smudgey) [inline test/uk/gov/hmrc/agentfirelationship/controllers/RelationshipControllerSpec.scala]: This doesn't look like it needs to change

File: test/uk/gov/hmrc/agentfirelationship/controllers/RelationshipControllerSpec.scala [hunk 2]
```diff
@@ -189,17 +191,19 @@ class RelationshipControllerSpec extends UnitSpec with MockitoSugar with GuiceOn
       verify(mockMongoService, times(0)).findRelationships(any(), any(), any())(any())
     }
 
-    "return Status: OK for deleting a record as a client" in {
+    "return Status: OK for deleting multiple records as a client" in {
```
Comment (Smudgey) [inline test/uk/gov/hmrc/agentfirelationship/controllers/RelationshipControllerSpec.scala]: These tests have been changed to work for deleting multiple relationships at once, which is the new functionality. However, this should be added as new additional tests, not overwriting the pre-existing tests for deleting a single relationship

File: conf/app.routes [hunk 3]
```diff
@@ -1,6 +1,9 @@
 # microservice specific routes
 
 
+GET           /relationships/getRelationships/service/:service/clientId/:clientId                     @uk.gov.hmrc.agentfirelationship.controllers.RelationshipController.findClientRelationships(service: String, clientId: String)
+DELETE        /relationships/de-auth-agents/service/:service/clientId/:clientId                       @uk.gov.hmrc.agentfirelationship.controllers.RelationshipController.deleteClientRelationships(service: String,clientId: String)
```
Comment (Smudgey) [inline conf/app.routes]: If you can, get these URLs run by Charlie

File: app/uk/gov/hmrc/agentfirelationship/services/RelationshipMongoService.scala [hunk 4]
```diff
@@ -34,22 +35,36 @@ class RelationshipMongoService @Inject()(mongoComponent: ReactiveMongoComponent)
...
-        "clientId" -> clientId)
-        .map(result => if (result.n == 0) false else result.ok)
+    remove(
+      "arn" -> arn,
+      "service" -> service,
+      "clientId" -> clientId)
+      .map(result => if (result.n == 0) false else result.ok)
+  }
+
+  def deleteRelationships(service: String, clientId: String)(implicit ec: ExecutionContext): Future[Boolean] = {
```
Comment (Smudgey) [inline app/uk/gov/hmrc/agentfirelationship/services/RelationshipMongoService.scala]: Can this be re-named to be clearer that this is deleting all relationships for a specific client?

File: test/uk/gov/hmrc/agentfirelationship/controllers/RelationshipControllerSpec.scala (hunk 2 above)
Comment (arkowalski) [reply test/uk/gov/hmrc/agentfirelationship/controllers/RelationshipControllerSpec.scala]: Ive adapted the tests that were only client based, as the functionality that was used before is no longer in use and only what I have added will be.

**Pull request closed without merging**